consumer/src/pkce/pkce_service.e

```diff
@@ -0,0 +1,166 @@
+note
+	description: "Object implementing PKCE service by OAuth Client"
+	date: "$Date$"
+	revision: "$Revision$"
+	EIS: "name", "src=https://tools.ietf.org/html/rfc7636#section-4.1", "protocol=URI"
+
+class
+	PKCE_SERVICE
+
+inherit
+
+	SHARED_BASE64
+
+create
+	make
+
+feature {NONE} -- Initialization
+
+	make (a_octets: INTEGER)
+			-- Create a random number generator using `a_number_of_octects`.
+			-- it recommended to use 32-octects.
+			--|   NOTE: The code verifier SHOULD have enough entropy to make it
+			--|   impractical to guess the value.  It is RECOMMENDED that the output of
+			--|   a suitable random number generator be used to create a 32-octet
+			--|   sequence.  The octet sequence is then base64url-encoded to produce a
+			--|   43-octet URL safe string to use as the code verifier.	
+		do
+			octet_length := a_octets
+		ensure
+			octet_length_set: octet_length = a_octets
+		end
+
+feature -- Access
+
+	octet_length: INTEGER
+			-- octect length.
+
+
+feature -- Generate PKCE
+
+	generate_pkce_sha256: PKCE
```

Comment:
name does not follow style guide line

it could be  "new_pkce_sha256"

Reply:
fixed.